solv/tdnfrepo.c

```diff
@@ -150,14 +150,52 @@ SolvLoadRepomdUpdateinfo(
     goto cleanup;
 }
 
+uint32_t
+SolvLoadRepomdOther(
+    Repo* pRepo,
+    const char* pszOther
+    )
+{
+    uint32_t dwError = 0;
+    FILE *fp = NULL;
+    if( !pRepo || IsNullOrEmptyString(pszOther))
+    {
+        dwError = ERROR_TDNF_INVALID_PARAMETER;
+        BAIL_ON_TDNF_LIBSOLV_ERROR(dwError);
+    }
+
+    fp = solv_xfopen(pszOther, "r");
+    if (fp == NULL)
+    {
+        dwError = ERROR_TDNF_SOLV_IO;
+        BAIL_ON_TDNF_LIBSOLV_ERROR(dwError);
+    }
+    if (repo_add_rpmmd(pRepo, fp, 0, REPO_EXTEND_SOLVABLES))
+    {
+        dwError = ERROR_TDNF_SOLV_IO;
+        BAIL_ON_TDNF_LIBSOLV_ERROR(dwError);
+    }
+cleanup:
+    if(fp != NULL)
+    {
+        fclose(fp);
+    }
+    return dwError;
+
+error:
```

Comment:
get rid of this error, it's not doing anything

We can improvise BAIL_ON_TDNF_ERROR macro a bit.
```
#define BAIL_ON_TDNF_ERROR(err, label) do { if (err) goto label; while (0)
```

Reply:
That's not making it more readable.